solutions/ha-nva/layer-7-ingress/README.md

```diff
@@ -0,0 +1,106 @@
+# Highly Available Ingress with layer 7 NVAs
+
+This sample template will deploy an Azure Application Gateway in front of two simulated network virtual appliance (NVA) VMs.  The NVA VMs will forward requests on port 80 to a Web-tier Azure Load Balancer, which will distribute traffic across two back-end VMs running a simple web server.
+
+For more information on this deployment and other highly available NVA options, see [Deploy highly available NVAs](https://docs.microsoft.com/azure/architecture/reference-architectures/dmz/nva-ha).
+
+![Highly Available Ingress with layer 7 NVAs architectural diagram.](l7-ingress.png)
+
+This sample architecture leverages concepts from the [Example Parameterized Deployment With Linked Templates](https://azure.microsoft.com/resources/templates/301-parameterized-linked-templates/) and [Create ssh-keys and store in Key Vault](https://azure.microsoft.com/resources/templates/201-deployment-script-ssh-key-gen/) samples.
+
+## Deploy this template to Azure
+
+*Note: If you fork this repository, you will need to modify the link in [README.md](README.md) to point to your repo.  If you create a separate branch for testing, you will have to include a change to this link to point to your branch as well. You must include a URL-encoded link to the raw [azuredeploy.json](azuredeploy.json) file after `/uri/` in the link defined for the deployment button. You should also change the default value of `_artifactsLocation`.*
```

Comment:
It seems like the artifacts location would also require updating in the main tempalte. May be worth noting here or have you considered consolidating down into a single tempalte file to work around?

Reply:
I had this in the last sentence above, but I added some text to make it more clear.